nomenclature/definition.py

```diff
@@ -40,6 +53,33 @@ def __init__(self, path, dimensions=["region", "variable"]):
         if empty:
             raise ValueError(f"Empty codelist: {', '.join(empty)}")
 
+        # check that any variable region-processing-rename mappings are valid
+        if "variable" in self.dimensions:
+            items = [
+                (name, attrs)
+                for (name, attrs) in self.variable.items()
+                if "region-aggregation" in attrs
+            ]
+            for (name, attrs) in items:
+                # ensure that there no pyam-aggregation-kwargs and
+                conflict_args = [i for i in attrs if i in PYAM_AGG_KWARGS]
+                if conflict_args:
+                    raise VariableRenameArgError(
+                        variable=name,
+                        file=attrs["file"],
+                        args=conflict_args,
+                    )
+
+                # ensure that mapped variables are defined in the nomenclature
+                rename_attrs = CodeList.from_list(
```

Comment:
I think this should also work without having to create another `CodeList` potentially eliminating the need for `CodeList.from_list`:

```suggestion
                invalid = [var_name for var in attrs['region-aggregation] for var_name in var if var_name not in self.variable]
```

Reply:
One additional benefit of casting to a `CodeList` is that this automatically guards against duplicate keys in the list. And I think "cast to a CodeList" is easier to grasp than the nested list comprehension...

Reply:
Ah yes, that safe guard is a fair point. I like the change to adding a pydantic validator to `CodeList`.
Could this validator also be used for the 'normal' `from_directory`?

Reply:
I'm working on it...

Reply:
Very nice.